### PR TITLE
feat(node): change sybil rejection to /14

### PR DESCRIPTION
**Pull request by clbartoli**
This PR changes the sybil rejection from 16 bits to a parameter from the config file

### CODE REVIEW

**Comment by aesedepece** [comment]
As per CIDR notation, the `X.0.0.0` mask corresponds to `/8`, not `/24`.

File: p2p/src/peers/mod.rs [hunk 1]
```diff
@@ -490,6 +490,22 @@ pub fn split_socket_addresses(socket_addr: &SocketAddr) -> (Vec<u8>, Vec<u8>, Ve
     }
 }
 
+/// Returns the /8 IP range
+pub fn get_range_address(socket_addr: &SocketAddr) -> Vec<u8> {
```
Comment (tmpolaczyk) [inline p2p/src/peers/mod.rs:494]: It would be nice to have a range argument here so that next time we want to change it it's just a one line change, but I guess it's better to do that in a separate PR?

```suggestion
pub fn get_range_address(socket_addr: &SocketAddr, range: u8) -> Vec<u8> {
```

Comment (lrubiorod) [reply p2p/src/peers/mod.rs:494]: I agree, it could be better in a separate PR

**Comment by mariocao** [comment]
Initially, we talked about increasing the permissiveness only by a few bits (e.g. 2 bits). Maybe we should discuss further if one whole octet is "too much" compared to the current Sybil rejection state.

File: witnet.toml [hunk 2]
```diff
@@ -114,9 +114,9 @@ known_peers = [
 outbound_limit = 8
 # Period for opening new peer connections while the current number of peers is lower than `outbound_limit`.
 bootstrap_peers_period_seconds = 1
-# Reject (tarpit) inbound connections coming from addresses in the same /16 IP range, so as to prevent sybil peers from
+# Reject (tarpit) inbound connections coming from addresses in the same /x IP range, so as to prevent sybil peers from
 # monopolizing our inbound capacity (128 by default).
-reject_sybil_inbounds = true
+reject_sybil_inbounds = 14
```
Comment (tmpolaczyk) [inline witnet.toml:119]: This change breaks existing `witnet.toml` files, and requires manual intervention to fix.

I propose to keep `true` as a legacy option and alias it to `16`.

Comment (mariocao) [reply witnet.toml:119]: Makes sense

Comment (mariocao) [reply witnet.toml:119]: Maybe change the name of the param is better :)

File: config/src/config.rs [hunk 3]
```diff
@@ -236,10 +236,14 @@ pub struct Connections {
     ))]
     pub bucketing_ice_period: Duration,
 
-    /// Reject (tarpit) inbound connections coming from addresses in the same /16 IP range, so as
+    /// Reject (tarpit) inbound connections coming from addresses in the same /x IP range, so as
     /// to prevent sybil peers from monopolizing our inbound capacity (128 by default).
     pub reject_sybil_inbounds: bool,
 
+    /// Limit range to reject (tarpit) inbound connections coming from addresses in the same /x IP range, so as
+    /// to prevent sybil peers from monopolizing our inbound capacity (128 by default).
```
Comment (mariocao) [inline config/src/config.rs:244]: The default changed :smile: 
```suggestion
    /// to prevent sybil peers from monopolizing our inbound capacity (128 by default).
```

File: witnet.toml (hunk 2 above)
Comment (drcpu-github) [reply witnet.toml]: Create two params? `reject_sybil_inbounds` set to `true` or `false` and `reject_sybil_inbounds_bits` set to an integer value? That would be a non-breaking change (I guess), and IMHO be more clear?

File: p2p/src/peers/mod.rs [hunk 4]
```diff
@@ -490,6 +490,29 @@ pub fn split_socket_addresses(socket_addr: &SocketAddr) -> (Vec<u8>, Vec<u8>, Ve
...
+/// Returns the /x IP range
+pub fn get_range_address(socket_addr: &SocketAddr, range: u8) -> [u8; 4] {
+    match socket_addr {
+        SocketAddr::V4(addr) => {
+            let ip = addr.ip().octets();
+
+            let ip_bytes = u32::from(ip[0]) << 24
+                | (u32::from(ip[1])) << 16
+                | (u32::from(ip[2])) << 8
+                | (u32::from(ip[3]));
```
Comment (tmpolaczyk) [inline p2p/src/peers/mod.rs:502]: Is this `u32::from_be_bytes`?

File: config/src/config.rs [hunk 5]
```diff
@@ -236,10 +236,14 @@ pub struct Connections {
...
     pub bucketing_ice_period: Duration,
 
-    /// Reject (tarpit) inbound connections coming from addresses in the same /16 IP range, so as
-    /// to prevent sybil peers from monopolizing our inbound capacity (128 by default).
+    /// Reject (tarpit) inbound connections coming from addresses in the same /x IP range, so as
+    /// to prevent sybil peers from monopolizing our inbound capacity (/14 by default).
     pub reject_sybil_inbounds: bool,
 
+    /// Limit range to reject (tarpit) inbound connections coming from addresses in the same /x IP range, so as
+    /// to prevent sybil peers from monopolizing our inbound capacity.
```
Comment (tmpolaczyk) [inline config/src/config.rs:244]: This comment is a bit hard to understand.

**Review by aesedepece — APPROVED** [review]

**Review by aesedepece — CHANGES_REQUESTED** [review]
-

File: witnet.toml (hunk 2 above)
Comment (mariocao) [reply witnet.toml]: yep! exactly that! :clap:

**Review by mariocao — APPROVED** [review]

**Review by lrubiorod — APPROVED** [review]

**Review by aesedepece — APPROVED** [review]
`utACK`

File: witnet.toml [hunk 6]
```diff
@@ -114,8 +114,8 @@ known_peers = [
 outbound_limit = 8
 # Period for opening new peer connections while the current number of peers is lower than `outbound_limit`.
 bootstrap_peers_period_seconds = 1
-# Reject (tarpit) inbound connections coming from addresses in the same /16 IP range, so as to prevent sybil peers from
-# monopolizing our inbound capacity (128 by default).
+# Reject (tarpit) inbound connections coming from addresses that are alike (i.e. by default having the fisrt 18 bits equal), so as to prevent sybil peers from
```
Comment (tmpolaczyk) [inline witnet.toml:117]: Typo: "fisrt", and also this line is too long

**Review by tmpolaczyk — APPROVED** [review]
